modules/library/main/src/test/java/org/geotools/filter/FilterFilterTest.java

```diff
@@ -109,6 +109,44 @@ public static void testWithoutFunction() throws Exception {
         assertEquals(FilterType.GEOMETRY_BBOX, Filters.getFilterType( f2));
         
     }
+	
+    public void testLikeMatchCase_v1_0()  throws Exception {
+        String filter = "<wfs:GetFeature service=\"WFS\" version=\"1.0.0\" " +
+             "outputFormat=\"GML2\" " +
+             "xmlns:topp=\"http://www.openplans.org/topp\" " +
+             "xmlns:wfs=\"http://www.opengis.net/wfs\" " +
+             "xmlns:ogc=\"http://www.opengis.net/ogc\" " +
+             "xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" " +
+             "xsi:schemaLocation=\"http://www.opengis.net/wfs http://schemas.opengis.net/wfs/1.0.0/WFS-basic.xsd\">" +
+             "<wfs:Query typeName=\"cite:Buildings\">" +
+               "<Filter xmlns=\"http://www.opengis.net/ogc\" xmlns:gml=\"http://www.opengis.net/gml\">" +
+                 "<PropertyIsLike wildCard=\"*\" singleChar=\".\" escapeChar=\"\\\" matchCase=\"false\">" +
+                   "<PropertyName>ADDRESS</PropertyName>" +
+                   "<Literal>* MAIN STREET</Literal>" +
+                 "</PropertyIsLike>" +
+               "</Filter>" +
+             "</wfs:Query>" + 
+           "</wfs:GetFeature>";
+        
+        StringReader reader = new StringReader( filter );
+        
+        InputSource requestSource = new InputSource(reader);
+        
+        // instantiante parsers and content handlers
+        MyHandler contentHandler = new MyHandler();
+        FilterFilter filterParser = new FilterFilter(contentHandler, null);
+        GMLFilterGeometry geometryFilter = new GMLFilterGeometry(filterParser);
+        GMLFilterDocument documentFilter = new GMLFilterDocument(geometryFilter);
+        
+        // read in XML file and parse to content handler
+        SAXParserFactory factory = SAXParserFactory.newInstance();
+        SAXParser parser = factory.newSAXParser();
+        ParserAdapter adapter = new ParserAdapter(parser.getParser());
+        adapter.setContentHandler(documentFilter);
+        adapter.parse(requestSource);
+        
+        assertEquals(contentHandler.filters.size(),1);
```

Comment:
So the test now checks it's not throwing an exception anymore, could you also test the parsed filter also has matchCase = false?